src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -245,11 +245,13 @@ protected virtual void Initialize()
                     // Disable core storage 
                     ScriptConfig.HostConfig.StorageConnectionString = null;
                 }
-
-                // Create the lease manager that will keep handle the primary host blob lease acquisition and renewal 
-                // and subscribe for change notifications.
-                _blobLeaseManager = BlobLeaseManager.Create(storageString, TimeSpan.FromSeconds(15), ScriptConfig.HostConfig.HostId, TraceWriter);
-                _blobLeaseManager.HasLeaseChanged += BlobLeaseManagerHasLeaseChanged;
+                else if (ScriptConfig.RoleDetectionEnabled)
+                {
```

Comment:
If you've set RoleDetectionEnabled to true but not provided a ConnectionString, should we throw there instead? It feels like you've asked us to do something that we can't do and maybe a descriptive error would help someone diagnose the problem.

---
In reply to: [72854143](https://github.com/Azure/azure-webjobs-sdk-script/pull/536#discussion_r72854143) [](ancestors = 72854143)